MySQL server has gone away after a scheduler task is finished

Hi everyone.

When I run a scheduler task on the shell I get a
```
[ERROR] An exception occurred while executing a query: MySQL server has gone away
Exception in scheduler task #4 (Aimeos Shop scheduler)
Class: Aimeos\Aimeos\Scheduler\Task\Typo6 
File: /var/www/html/vendor/doctrine/dbal/src/Driver/API/MySQL/ExceptionConverter.php:93
```
The task itself is successful. The error gets thrown as soon as TYPO3 is back in control and tries to create this prepared statement
```sql
SELECT * FROM `tx_scheduler_task` WHERE (`uid` = ?) AND (`deleted` = ?)
```
inside the SchedulerTaskRepository.

The scheduler task then still has the `running` state.

**Environment**

1. TYPO3 13.4.19
2. PHP 8.2.x
3. Aimos Extension 25.10.2


Tobi

--- a/Classes/Scheduler/Base.php
+++ b/Classes/Scheduler/Base.php
@@ -39,9 +39,6 @@ class Base
         $context = self::context($tsconf, $conf, $pid);
         $process = $context->process();
 
-        // Reset before child processes are spawned to avoid lost DB connections afterwards
-        GeneralUtility::makeInstance('TYPO3\CMS\Core\Database\ConnectionPool')->resetConnections();
-
         $manager = \Aimeos\MShop::create($context, 'locale');
 
         foreach (self::getSiteItems($context, $sites) as $siteItem) {
@@ -58,6 +55,8 @@ class Base
             $tmplPaths = Aimeos\Base::aimeos()->getTemplatePaths('controller/jobs/templates', $theme);
             $context->setView(Aimeos\Base::view($context, self::getRouter($pid), $tmplPaths));
 
+            // Reset before child processes are spawned to avoid lost DB connections afterwards
+            GeneralUtility::makeInstance('TYPO3\CMS\Core\Database\ConnectionPool')->resetConnections();
             foreach ($jobs as $jobname) {
                 $fcn = function($context, $aimeos, $jobname) {
                     \Aimeos\Controller\Jobs::create($context, $aimeos, $jobname)->run();

